go/vt/vtctl/vtctl.go

```diff
@@ -3399,21 +3400,26 @@ func commandApplyVSchema(ctx context.Context, wr *wrangler.Wrangler, subFlags *p
 
 	// Log unknown Vindex params as warnings.
 	var vdxNames []string
+	var unknownParams []string
 	for name := range ksVs.Vindexes {
 		vdxNames = append(vdxNames, name)
 	}
 	sort.Strings(vdxNames)
 	for _, name := range vdxNames {
 		vdx := ksVs.Vindexes[name]
 		if val, ok := vdx.(vindexes.ParamValidating); ok {
-			for _, param := range val.UnknownParams() {
-				wr.Logger().Warningf("Unknown parameter in vindex %s: %s", name, param)
-			}
+			unknownParams = append(unknownParams, fmt.Sprintf("%s (%s)", name, strings.Join(val.UnknownParams(), ", ")))
 		}
 	}
+	if len(unknownParams) > 0 {
+		wr.Logger().Warningf("Unknown parameter(s) in vindex(es): %s", strings.Join(unknownParams, "; "))
+	}
 
-	if *dryRun {
+	if *dryRun || (*strict && len(unknownParams) > 0) {
 		wr.Logger().Printf("Dry run: Skipping update of VSchema\n")
```

Comment:
Maybe we can keep the strict/unknown params logic separate from dry run? For example, if the user didn't specify dry run, but did specify strict, this log line about "Dry run: Skipping update of VSchema\n" would still be output.